roxie/ccd/ccdqueue.cpp

```diff
@@ -1988,6 +1988,31 @@ class RoxieReceiverBase : implements IRoxieOutputQueueManager, public CInterface
 #pragma warning ( disable: 4355 )
 #endif
 
+static void throwPacketTooLarge(IRoxieQueryPacket *x, unsigned maxPacketSize)
+{
+    StringBuffer t;
+    unsigned traceLength = x->getTraceLength();
+    if (traceLength)
+    {
+        const byte *traceInfo = x->queryTraceInfo();
+        unsigned char loggingFlags = *traceInfo;
+        if (loggingFlags & LOGGING_FLAGSPRESENT) // should always be true.... but this flag is handy to avoid flags byte ever being NULL
+        {
+            traceInfo++;
+            traceLength--;
+            if (loggingFlags & LOGGING_TRACELEVELSET)
+            {
+                traceInfo++;
+                traceLength--;
```

Comment:
Could traceLength ever be 0 before this decrement ?

Reply:
Only if packets were corrupt. There's similar code at the agent end interpreting the received data...